daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/NodeInfo.scala

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.transaction
+
+/** Trait for extracting information from an abstract node.
+  * Used for sharing the implementation of common computations
+  * over nodes and transactions.
+  */
+trait NodeInfo[PartyRep] {
+
+  /** The node kind, e.g. create, consuming exercise etc. */
+  def kind: NodeKind
+
+  /** The signatories of the contract associated with the node. */
+  def signatories: Set[PartyRep]
```

Comment:
I'm not a fan of adding members to an interface if not all implementations sensibly can implement it. According to the description, a positive key lookup should also provide signatories and stakeholders.

Reply:
Well yes they should be provided by it, but it’s a bigger refactoring to add them. For negative lookups we wouldn’t be able to provide them. Perhaps better to just extend NodeKind to carry the payload?

Reply:
Also, creates don't have actors. Perhaps we should use some other notions instead?